tests/test_header_guard.py

```diff
@@ -0,0 +1,177 @@
+import textwrap
+from pathlib import Path
+import pytest
+import header_guard as hg
+
+
+# ---------- Unit tests ----------
+
+def test_is_header_file():
+    assert hg.is_header_file(Path("a.hpp"))
+    assert not hg.is_header_file(Path("a.cpp"))
+
+
+def test_macro_from_rel_path_basic():
+    assert hg.macro_from_rel_path(Path("src/lib/vec3.hpp")) == "SRC_LIB_VEC3_HPP_"
+
+
+def test_macro_from_rel_path_normalizes():
+    p = Path(r"include\\foo+bar/baz qux.hh")
+    assert hg.macro_from_rel_path(p) == "INCLUDE_FOO_BAR_BAZ_QUX_HH_"
+
+
+def test_leading_comments_span_block():
```

Comment:
To be safe, please also add a test with a block comment with more than two lines.